api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -669,3 +676,38 @@ func (m *AzureManagedControlPlane) validateIntegerStringGreaterThanZero(input *s
 
 	return allErrs
 }
+
+// validateIdentityProfile validates an AutoScalerProfile.
+func (m *AzureManagedControlPlane) validateIdentityProfile(_ client.Client) error {
+	var allErrs field.ErrorList
+
+	if m.Spec.IdentityProfile != nil {
+		if m.Spec.IdentityProfile.Type == ManagedControlPlaneIdentityTypeUserAssigned {
+			if m.Spec.IdentityProfile.UserAssignedIdentityResourceID == "" {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "IdentityProfile", "UserAssignedIdentityResourceID"), m.Spec.IdentityProfile.UserAssignedIdentityResourceID, "cannot be empty if IdentityProfile.Type is UserAssigned"))
+			}
+			if m.Spec.IdentityProfile.UserAssignedIdentityPrincipalID == nil {
```

Comment:
have you tested this? I'm guessing you're following https://learn.microsoft.com/en-us/azure/aks/use-managed-identity ?

principal ID and client ID are _outputs_ there I thought, iirc you only need the UAMI ARM ID to pass to AKS. could be wrong.

Reply:
It's a bit of a mystery!

E0619 20:02:48.596549       1 controller.go:329]  "msg"="Reconciler error" "error"="error creating AzureManagedControlPlane capz-e2e-rwuh71/capz-e2e-rwuh71-aks: failed to reconcile AzureManagedControlPlane service managedcluster: failed to create resource capz-e2e-rwuh71-aks/capz-e2e-rwuh71-aks (service: managedcluster): Code=\"InvalidIdentityValues\" Message=\"Invalid value for the identities '/subscriptions/<redacted>/resourcegroups/capz-ci/providers/Microsoft.ManagedIdentity/userAssignedIdentities/umd-test'. The 'UserAssignedIdentities' property keys should only be empty null during PATCH requests.\"" "AzureManagedControlPlane"={"name":"capz-e2e-rwuh71-aks","namespace":"capz-e2e-rwuh71"} "controller"="azuremanagedcontrolplane" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedControlPlane" "name"="capz-e2e-rwuh71-aks" "namespace"="capz-e2e-rwuh71" "reconcileID"="b8fe5ded-9132-42cc-89ea-ae1621f4dbcf"

The above is in response to only passing in the URI as the key as one does here:

- https://github.com/Azure/azure-sdk-for-go/blob/v67.0.0/services/containerservice/mgmt/2022-03-01/containerservice/models.go#L1770-L1780

Your assertion that the principalID and clientID are output does check out with the type description here:

- https://github.com/Azure/azure-sdk-for-go/blob/v67.0.0/services/containerservice/mgmt/2022-03-01/containerservice/models.go#L1794-L1800

I'll try hacking around with a pointer to an empty `ManagedClusterIdentityUserAssignedIdentitiesValue` literal as the value instead of `nil` and see if it still complains about PATCH (or complains about anything), stand by....

Reply:
yep that worked!